File: 0283-move-zeroes/0283-move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int s = nums.size();
        int k,j,index=s-1;
        for(int i=s-1;i>=0;i--)
        {
            if(nums[i]==0)
            {
                for(j=i;j<index;j++)
                {
                    nums[j]=nums[j+1];
                }
                nums[index]=0;
                index--;
            }
        }
    }
};

File: 0283-move-zeroes/0283-move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int index=0;
        for(int i=0;i<nums.length;i++)
        {
            if(nums[i]!=0)
            {
                
                int t = nums[i];
                nums[i] = nums[index];
                nums[index] = t;
                    index++;
            }
        }
    }
}